modules/cli/app.go

```diff
@@ -81,43 +111,185 @@ func (app *App) GetAttr(name string) (object.Object, bool) {
+		}
+		app.value.Version = str.Value()
+	case "description":
+		str, ok := value.(*object.String)
+		if !ok {
+			return object.TypeErrorf("type error: expected string, got %s", value.Type())
+		}
+		app.value.Description = str.Value()
+	case "default_command":
+		str, ok := value.(*object.String)
+		if !ok {
+			return object.TypeErrorf("type error: expected string, got %s", value.Type())
+		}
+		app.value.DefaultCommand = str.Value()
+	case "enable_bash_completion":
+		b, ok := value.(*object.Bool)
+		if !ok {
+			return object.TypeErrorf("type error: expected bool, got %s", value.Type())
+		}
+		app.value.EnableBashCompletion = b.Value()
+	case "hide_help":
+		b, ok := value.(*object.Bool)
+		if !ok {
+			return object.TypeErrorf("type error: expected bool, got %s", value.Type())
+		}
+		app.value.HideHelp = b.Value()
+	case "hide_help_command":
+		b, ok := value.(*object.Bool)
+		if !ok {
+			return object.TypeErrorf("type error: expected bool, got %s", value.Type())
+		}
+		app.value.HideHelpCommand = b.Value()
+	case "hide_version":
+		b, ok := value.(*object.Bool)
+		if !ok {
+			return object.TypeErrorf("type error: expected bool, got %s", value.Type())
+		}
+		app.value.HideVersion = b.Value()
+	default:
+		return object.TypeErrorf("type error: %s object has no attribute %q", APP, name)
+	}
+	return nil
+}
+
 func NewApp(ctx context.Context, opts *object.Map) (*App, error) {
 	app := &ucli.App{}
 
-	var err error
-	app.Name, err = getMapStr(opts, "name")
-	if err != nil {
-		return nil, err
-	}
-	app.HelpName, err = getMapStr(opts, "help_name")
-	if err != nil {
-		return nil, err
-	}
-	app.Usage, err = getMapStr(opts, "usage")
-	if err != nil {
-		return nil, err
+	// Handle string fields
+	for _, field := range []string{
+		"name", "help_name", "usage", "usage_text", "args_usage",
+		"version", "description", "default_command",
+	} {
+		if opt := opts.Get(field); opt != object.Nil {
+			str, ok := opt.(*object.String)
+			if !ok {
+				return nil, object.TypeErrorf("type error: %s must be a string", field)
+			}
+			switch field {
+			case "name":
+				app.Name = str.Value()
+			case "help_name":
+				app.HelpName = str.Value()
+			case "usage":
+				app.Usage = str.Value()
+			case "usage_text":
+				app.UsageText = str.Value()
+			case "args_usage":
+				app.ArgsUsage = str.Value()
+			case "version":
+				app.Version = str.Value()
+			case "description":
+				app.Description = str.Value()
+			case "default_command":
+				app.DefaultCommand = str.Value()
+			}
+		}
 	}
-	app.UsageText, err = getMapStr(opts, "usage_text")
-	if err != nil {
-		return nil, err
+
+	// Handle boolean fields
+	for _, field := range []string{
+		"args", "enable_bash_completion", "hide_help",
+		"hide_help_command", "hide_version",
+	} {
+		if opt := opts.Get(field); opt != object.Nil {
+			b, ok := opt.(*object.Bool)
+			if !ok {
+				return nil, object.TypeErrorf("type error: %s must be a boolean", field)
```

Comment:
All the other error messages always refer to the type as "bool":

```suggestion
				return nil, object.TypeErrorf("type error: %s must be a bool", field)
```